segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/Ledgers.java

```diff
@@ -67,11 +74,17 @@ static LedgerHandle create(BookKeeper bookKeeper, BookKeeperConfig config) throw
      * @return A LedgerHandle for the newly opened ledger.
      * @throws DurableDataLogException If an exception occurred. The causing exception is wrapped inside it.
      */
-    static LedgerHandle openFence(long ledgerId, BookKeeper bookKeeper, BookKeeperConfig config) throws DurableDataLogException {
+    static ReadHandle openFence(long ledgerId, BookKeeper bookKeeper, BookKeeperConfig config) throws DurableDataLogException {
         try {
             return Exceptions.handleInterruptedCall(
-                    () -> bookKeeper.openLedger(ledgerId, LEDGER_DIGEST_TYPE, config.getBKPassword()));
-        } catch (BKException bkEx) {
+                    () -> FutureUtils.result(bookKeeper
```

Comment:
Do we still need to handle interrupted here?

Reply:
yes I will do

Reply:
Agreed. We should file an issue to convert the methods in this class to use `CompletableFuture` now that the underlying APIs support that. We shouldn't do that as part of this effort as there'd be a lot of upstream code to update.

Reply:
#4433